### PR TITLE
Very basic version of viewing a file

### LINKED ISSUE #18: Load a buffer with :edit or :view
As someone who uses a text editor, I need to be able to open a file, so that I can, er, at _least_ view the text.

### PR DESCRIPTION
**Pull request by Kazark**
This enables loading a file with `;edit` or `;view`. It is fragmentary but I don't know where to take it from here without building out some other things.

Once this is done, I need to build and create the first release, `0.0.0`!

This fixes #18.

### CODE REVIEW

File: src/Void.Core/Filesystem.fs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    type LinesOrFailure =
+        | Lines of string seq
+        | Failure of Error
+
+    let private readLines path =
+        try
+            File.ReadLines path |> Lines
+        with
+        | :? UnauthorizedAccessException ->
```
Comment (mjgpy3) [inline src/Void.Core/Filesystem.fs:15]: Does `:? UnauthorizedAccessException` mean an error in the hierarchy of `UnauthorizedAccessException`?

File: src/Void.Core/NormalMode.fs [hunk 2]
```diff
@@ -18,7 +18,6 @@ module NormalMode =
             [KeyPress.ShiftZ; KeyPress.ShiftQ], Command.QuitWithoutSaving
             [KeyPress.ShiftZ; KeyPress.ShiftA], Command.QuitAll
             [KeyPress.G; KeyPress.Q; KeyPress.Q], Command.FormatCurrentLine
-            [KeyPress.F12], Command.ViewTestBuffer
```
Comment (mjgpy3) [inline src/Void.Core/NormalMode.fs:21]: The test buffer was cool, but it's nice to see it go :)

File: src/Void.Core/Filesystem.fs [hunk 3]
```diff
@@ -0,0 +1,33 @@
+﻿namespace Void.Core
+
+module Filesystem =
+    open System
+    open System.IO
+
+    type LinesOrFailure =
```
Comment (mjgpy3) [inline src/Void.Core/Filesystem.fs:7]: Does F# have an `Either` type?

Comment (mjgpy3) [inline src/Void.Core/Filesystem.fs:7]: In Haskell We'd probably just say `Either [String] Error` at the point of usage.

File: src/Void.Core/Filesystem.fs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+        with
+        | :? UnauthorizedAccessException ->
+            Error.AccessToPathNotAuthorized path |> Failure
+
+    let handleCommand =
+        function
+        | Command.Edit fileId ->
+            match fileId with
+            | FileIdentifier.Path path ->
+                if File.Exists path
```
Comment (mjgpy3) [inline src/Void.Core/Filesystem.fs:23]: This logic is neato!

File: src/Void/Init.fs [hunk 5]
```diff
@@ -35,11 +35,13 @@ module Init =
             Channel [
```
Comment (mjgpy3) [inline src/Void/Init.fs:35]: Just read over this file. This code is awesome!

Comment (Kazark) [inline src/Void/Init.fs:35]: Haha, I know. I think I inadvertently became a microservices guy. Did you look at the `CommandBar` service? I love the way that played out in the specs---felt like I was hitting a perfect testing level, testing a sane amount of complexity and the tests very unlikely to be brittle.

File: src/Void.Core/Filesystem.fs (hunk 3 above)
Comment (Kazark) [inline src/Void.Core/Filesystem.fs:7]: Yeah, it does. I haven't convinced myself of which style I prefer.

File: src/Void.Core/Filesystem.fs (hunk 1 above)
Comment (Kazark) [inline src/Void.Core/Filesystem.fs:15]: Yeah, it's a type check. Interestingly enough, try/catch is sort of the one example of pattern matching in most imperative languages.

**Comment by Kazark** [comment]
@mjgpy3 Thanks for looking this stuff over! Feel free to merge!

File: src/Void.Core/Filesystem.fs (hunk 3 above)
Comment (mjgpy3) [inline src/Void.Core/Filesystem.fs:7]: Fair enough :)

File: src/Void.Core/Filesystem.fs (hunk 1 above)
Comment (mjgpy3) [inline src/Void.Core/Filesystem.fs:15]: Interesting, I hadn't though it of like that before!

`Exception e` is kind of like your catch-all case, huh?

Comment (Kazark) [inline src/Void.Core/Filesystem.fs:15]: Yep. It's rather crude pattern matching, but it sort of is a subset of it anyway.

**Pull request merged**